src/lightbox/downloadImage.js

```diff
@@ -35,27 +34,35 @@ const androidEnsureStoragePermission = async (): Promise<void> => {
 };
 
 export default async (src: string, auth: Auth): Promise<mixed> => {
-  const absoluteUrl = getFullUrl(src, auth.realm);
+  const fileDownloadUrl: string = await getFileDownloadUrl(src, auth);
 
   if (Platform.OS === 'ios') {
-    const delimiter = absoluteUrl.includes('?') ? '&' : '?';
-    const urlWithApiKey = `${absoluteUrl}${delimiter}api_key=${auth.apiKey}`;
-    return CameraRoll.saveToCameraRoll(urlWithApiKey);
+    try {
+      return CameraRoll.saveToCameraRoll(fileDownloadUrl);
+    } catch (err) {
+      openLink(fileDownloadUrl);
+
+      // Tell the caller that the image opened in the browser.
+      return 'browser';
```

Comment:
This doesn't feel like the right type, given that this function can also return real data.

I see we don't use that data in the call site in `LightboxActionSheet`. Is there somewhere else we're using it? If we're not using it, we could just reduce the return value to true/false.

Nope, we do use it -- in `shareImage.js`.  I bet that will need logic to handle this change in this function's API too, then.

Reply:
`shareImage` currently does not work on Android - both on CZO and *.zulipchat.con

Reply:
Agreed, the type is 
```js
Promise<{|
  url: string,
  openedInBrowser: boolean,
|}>
``` 
now, and I've made changes so that this is handled properly.